api/helpers/uploadedFileErrorChecks.js

```diff
@@ -293,9 +342,24 @@ function formatFileSizeMb(bytes, decimalPlaces = 1) {
     return `${(bytes / 1_000_000).toFixed(decimalPlaces)}Mb`;
 }
 
+class ErrorAndDeleteFile extends Error {
+    constructor(message, req, file) {
+        super(message);
+        removeFileFromSessionAndDelete(req, file);
+    }
+}
```

Comment:
the rule of thumb is if a function (or class in this case) has "and" in the name it's probably got too many responsibilities. 

Errors should not do anything apart from report the error. For example, what happens when removeFileFromSessionAndDelete fails? why is an error spurting another possible error? 😨 

should go something like this

```.js 
async function addUnsubmittedTag(file, req) {
  try {
  // ...
   await s3.send(new PutObjectTaggingCommand(params));
   sails.log.info(`Only UNSUBMITTED tag added to ${fileStorageName}`);
  } catch (error) {
    throw Error(`STAYONPAGE: ${error}`)
  }
}

try { 
  addCleanAndUnsubmittedTagsToFile(..)
} catch (error) { //this is the STAYONPAGE error. every error from addCleanAndUnsubmittedTagsToFile will return this error

 //optionally check the message/instanceof the error if you want to handle the errors differently

 removeFileFromSessionAndDelete(req, file)
}
```

Reply:
This is a really good point I didn't think of this.

I could be completely wrong but what I take from your example is that the function to remove files should be slightly higher up in the chain?